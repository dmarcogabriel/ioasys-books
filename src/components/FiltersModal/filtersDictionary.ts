export const CATEGORY = {
  biographies: 'Biografias',
  collections: 'Coleções',
  behavior: 'Comportamento',
  tales: 'Contos',
  'literary-criticism': 'Literário - Crítica',
  scienceFiction: 'Ficção Científica',
  folklore: 'Folclóre',
  genealogy: 'Genealogia',
  humor: 'Humor',
  children: 'Infantil',
  games: 'Jogos',
  newspapers: 'Jornais',
  'brazilian-literature': 'Literatura Brasileira',
  'foreign-literature': 'Literatura Internacional',
  'rare-books': 'Livros Raros',
  manuscripts: 'Manuscritos',
  poetry: 'Poesia',
  'another-subjects': 'Outros Assuntos',
};
