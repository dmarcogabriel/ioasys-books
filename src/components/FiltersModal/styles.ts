import styled from 'styled-components/native';
import {StyleSheet} from 'react-native';
import {ThemeProps} from '../../@types/theme.interface';

export default StyleSheet.create({
  modal: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 16,
  },
});

export const FiltersModalContainer = styled.ScrollView`
  background: ${({theme}: ThemeProps) => theme.colors.light};
  border-radius: 4px;
  padding: 16px 16px 24px;
  width: 100%;
`;

export const FiltersModalHeader = styled.View`
  align-items: flex-end;
`;

export const CloseButtonWrapper = styled.TouchableOpacity`
  border-radius: 100px;
  border: 1px solid #33333333;
  padding: 9px;
`;

export const FiltersModalFooter = styled.View`
  align-items: center;
  justify-content: center;
`;

export const FilterButton = styled.TouchableOpacity`
  border: 1px solid ${({theme}: ThemeProps) => theme.colors.primary};
  border-radius: 20px;
  padding: 8px 20px;
  align-items: center;
  justify-content: center;
  margin: 32px 0;
`;

export const FilterButtonText = styled.Text`
  color: ${({theme}: ThemeProps) => theme.colors.primary};
  font-size: 16px;
`;
