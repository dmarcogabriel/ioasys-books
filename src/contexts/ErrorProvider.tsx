import React, {createContext} from 'react';
import {UNAUTHORIZED_STATUS} from '../constants';
import Toast from 'react-native-toast-message';

type ErrorData = {
  message: string;
};

export interface ApiError {
  errors: ErrorData;
}

interface ErrorContextValue {
  handleError: (
    error: ApiError,
    status: number,
    logoutCallback: () => Promise<void>,
  ) => Promise<void>;
}

export const ErrorContext = createContext<ErrorContextValue>({
  handleError: async () => {},
});

interface ErrorProviderProps {
  children: React.ReactNode;
}

export const ErrorProvider = ({children}: ErrorProviderProps): JSX.Element => {
  const handleError = async (
    error: ApiError,
    status: number,
    logoutCallback: () => Promise<void>,
  ): Promise<void> => {
    let errorMessage: string = '';
    if (status === UNAUTHORIZED_STATUS) {
      await logoutCallback();
      errorMessage = 'Sess√£o expirada';
    } else {
      Toast.show({
        position: 'bottom',
        text1: errorMessage || error.errors.message,
        type: 'error',
      });
    }
  };

  return (
    <ErrorContext.Provider value={{handleError}}>
      {children}
    </ErrorContext.Provider>
  );
};
