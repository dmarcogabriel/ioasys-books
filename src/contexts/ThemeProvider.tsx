import React from 'react';
import {ThemeProvider as SCThemeProvider} from 'styled-components/native';

interface ThemeProviderProps {
  children: React.ReactNode;
}

export const ThemeProvider = ({children}: ThemeProviderProps): JSX.Element => {
  const theme = {
    colors: {
      primary: '#B22E6F',
      dark: '#333333',
      light: '#ffffff',
      grey: '#999999',
    },
  };

  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};
