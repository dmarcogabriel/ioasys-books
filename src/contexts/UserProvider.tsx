import React, {createContext, useState, useEffect, useCallback} from 'react';
import {User} from '../@types/user.interface';
import {storeData, removeData, getData} from '../utils/storage';
import {AUTH_TOKEN_STORAGE_KEY} from '../constants';

interface UserContextValue {
  user: User | null;
  login: (user: User, token: string) => Promise<void>;
  logout: () => Promise<void>;
  token: string | null;
}

export const UserContext = createContext<UserContextValue>({
  user: null,
  token: null,
  login: async () => {},
  logout: async () => {},
});

interface UserProviderProps {
  children: React.ReactNode;
}

export const UserProvider = ({children}: UserProviderProps): JSX.Element => {
  const [user, setUser] = useState<User | null>(null);
  const [token, setToken] = useState<string | null>(null);
  const [isLoading, setIsLoading] = useState(true);

  const login = async (data: User, authToken: string): Promise<void> => {
    setUser(data);
    setToken(authToken);
    await storeData(AUTH_TOKEN_STORAGE_KEY, authToken);
  };

  const logout = async () => {
    await removeData(AUTH_TOKEN_STORAGE_KEY);
    setUser(null);
    setToken(null);
  };

  const loadToken = useCallback(async () => {
    setIsLoading(true);
    try {
      const authToken = await getData<string>(AUTH_TOKEN_STORAGE_KEY);
      if (authToken) {
        setToken(authToken);
      }
    } catch (error) {
      setIsLoading(false);
    }
    setIsLoading(false);
  }, []);

  useEffect(() => {
    loadToken();
  }, [loadToken]);

  return (
    <UserContext.Provider value={{user, token, login, logout}}>
      {!isLoading && children}
    </UserContext.Provider>
  );
};
