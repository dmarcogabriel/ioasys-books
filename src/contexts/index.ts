export * from './UserProvider';
export * from './ThemeProvider';
export * from './ErrorProvider';
