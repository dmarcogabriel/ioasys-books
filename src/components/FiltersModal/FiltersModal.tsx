import React, {useState} from 'react';
import RNModal from 'react-native-modal';
import MIcon from 'react-native-vector-icons/MaterialIcons';
import styles, {
  FiltersModalContainer,
  FiltersModalHeader,
  CloseButtonWrapper,
  FilterButton,
  FilterButtonText,
  FiltersModalFooter,
} from './styles';
import {FiltersChips, Option} from './components';
import {CATEGORY} from './filtersDictionary';

export interface FiltersModalProps {
  isVisible: boolean;
  onClose: (filters?: any) => void;
  yearOptions: string[];
  categoryOptions: string[];
  authorOptions: string[];
}

interface FiltersState {
  category: string | null;
  year: string | null;
  author: string | null;
}

type FieldName = 'category' | 'year' | 'author';

export const FiltersModal = ({
  isVisible,
  yearOptions = [],
  categoryOptions = [],
  authorOptions = [],
  onClose,
}: FiltersModalProps): JSX.Element => {
  const [filters, setFilters] = useState<FiltersState>({
    category: null,
    year: null,
    author: null,
  });
  const handleClose = (filtersData?: FiltersState) => {
    onClose(filtersData || null);
  };

  const handleChange = (fieldName: FieldName) => (option: string) => {
    if (filters[fieldName] === option) {
      setFilters({...filters, [fieldName]: null});
    } else {
      setFilters({...filters, [fieldName]: option});
    }
  };

  const defineOptions = (
    options: string[],
    dictionary?: {[key: string]: string},
  ): Option[] => {
    return options.map(option => ({
      name: dictionary ? dictionary[option] : option,
      value: option,
    }));
  };

  return (
    <RNModal
      testID="filtersModal"
      style={styles.modal}
      isVisible={isVisible}
      onBackdropPress={() => handleClose()}>
      <FiltersModalContainer>
        <FiltersModalHeader>
          <CloseButtonWrapper
            testID="closeButton"
            onPress={() => handleClose()}>
            <MIcon name="close" size={24} />
          </CloseButtonWrapper>
        </FiltersModalHeader>
        <FiltersChips
          testID="Category"
          title="Selecione a categoria"
          options={defineOptions(categoryOptions, CATEGORY)}
          value={filters.category}
          onChange={handleChange('category')}
        />
        <FiltersChips
          testID="Year"
          title="Selecione o ano"
          options={defineOptions(yearOptions)}
          value={filters.year}
          onChange={handleChange('year')}
        />
        <FiltersChips
          testID="Author"
          title="Selecione o autor"
          options={defineOptions(authorOptions)}
          value={filters.author}
          onChange={handleChange('author')}
        />
        <FiltersModalFooter>
          <FilterButton
            testID="filterButton"
            onPress={() => handleClose(filters)}>
            <FilterButtonText>Filtrar</FilterButtonText>
          </FilterButton>
        </FiltersModalFooter>
      </FiltersModalContainer>
    </RNModal>
  );
};
