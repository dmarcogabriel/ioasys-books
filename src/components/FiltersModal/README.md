# FiltersModal

This is an abstraction of `react-native-modal` lib it shows the books filters and return then 
when user closes it.

## Usage

```js
import {FiltersModal} from 'src/components';

...
  const [isVisible, setIsVisible] = useState(true);

  const options = {
    year: ['1992', '1998', '2000'],
    category: ['biographies', 'poetry'],
    author: ['Luke', 'Lea']
  };

  const handleClose = (filters) {
    if (filters) {
      // do something with filters
    }
  }

  <FiltersModal
    isVisible={isVisible}
    onClose={handleClose}
    yearOptions={options.year}
    categoryOptions={options.category}
    authorOptions={options.author}
  />
...
```

## API

| Prop         | Description                            | Type                      | Default   |
| ------------ | -------------------------------------- | ------------------------- | --------- |
| **`isVisible`**       | Controls modal visibility     | boolean                   | undefined |
| **`yearOptions`**     | Array of options for years    | string[]                  | undefined |
| **`categoryOptions`** | Array of options for category | string[]                  | undefined |
| **`authorOptions`**   | Array of options for author   | string[]                  | undefined |
| **`onClose`**         | Function executed when closes | function (callback): void | undefined |

#### onClose

`onClose` event execute a callback function when the modal closes, if user closes the modal by
pressing the Close button (x), it will execute the callback with params null. Otherwise if user
closes it by pressing the "Filtrar" button, it will execute the callback with the filters as 
params.