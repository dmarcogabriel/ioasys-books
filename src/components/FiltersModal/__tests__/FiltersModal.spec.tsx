import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import {FiltersModal, FiltersModalProps} from '../FiltersModal';
import {ThemeProvider} from '../../../contexts';
import renderer from 'react-test-renderer';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: mockNavigate,
  }),
}));

const ComponentWrapper = (props: FiltersModalProps) => (
  <ThemeProvider>
    <FiltersModal {...props} />
  </ThemeProvider>
);

let mockSelectedOption: any | null;
const mockOnClose = jest.fn((filters: any | null) => {
  mockSelectedOption = filters;
});

const mockYearOptions = ['1992', '1993', '1994'];
const mockCategoryOptions = ['category 1', 'category 2', 'category 3'];
const mockAuthorOptions = ['Author 1', 'Author 2', 'Author 3'];

beforeEach(() => {
  mockSelectedOption = null;
});

describe('components/FiltersModal', () => {
  it('should render correctly', () => {
    const {getByTestId} = render(
      <ComponentWrapper
        yearOptions={mockYearOptions}
        categoryOptions={mockCategoryOptions}
        authorOptions={mockAuthorOptions}
        onClose={mockOnClose}
        isVisible
      />,
    );
    const filtersModal = getByTestId('filtersModal');

    expect(filtersModal).toHaveProp('isVisible', true);
  });

  it('should render correctly not visible', () => {
    const {getByTestId} = render(
      <ComponentWrapper
        yearOptions={mockYearOptions}
        categoryOptions={mockCategoryOptions}
        authorOptions={mockAuthorOptions}
        onClose={mockOnClose}
        isVisible={false}
      />,
    );
    const filtersModal = getByTestId('filtersModal');

    expect(filtersModal).toHaveProp('isVisible', false);
  });

  it('should pass on press close button', () => {
    const {getByTestId} = render(
      <ComponentWrapper
        yearOptions={mockYearOptions}
        categoryOptions={mockCategoryOptions}
        authorOptions={mockAuthorOptions}
        onClose={mockOnClose}
        isVisible
      />,
    );
    fireEvent.press(getByTestId('closeButton'));

    expect(mockOnClose).toHaveBeenCalled();
    expect(mockSelectedOption).toBe(null);
  });

  it('should pass on press filter button', () => {
    const {getByTestId} = render(
      <ComponentWrapper
        yearOptions={mockYearOptions}
        categoryOptions={mockCategoryOptions}
        authorOptions={mockAuthorOptions}
        onClose={mockOnClose}
        isVisible
      />,
    );
    fireEvent.press(getByTestId('filterButton'));

    expect(mockOnClose).toHaveBeenCalled();
    expect(mockSelectedOption.category).toBe(null);
    expect(mockSelectedOption.year).toBe(null);
  });

  it('should pass on change filter', () => {
    const {getByTestId} = render(
      <ComponentWrapper
        yearOptions={mockYearOptions}
        categoryOptions={mockCategoryOptions}
        authorOptions={mockAuthorOptions}
        onClose={mockOnClose}
        isVisible
      />,
    );
    fireEvent.press(getByTestId('filtersChipsOptionButtonCategory2'));
    fireEvent.press(getByTestId('filtersChipsOptionButtonYear2'));
    fireEvent.press(getByTestId('filterButton'));

    expect(mockOnClose).toHaveBeenCalled();
    expect(mockSelectedOption.category).toBe('category 2');
    expect(mockSelectedOption.year).toBe('1993');
  });

  it('should pass on press same option', () => {
    const {getByTestId} = render(
      <ComponentWrapper
        yearOptions={mockYearOptions}
        categoryOptions={mockCategoryOptions}
        authorOptions={mockAuthorOptions}
        onClose={mockOnClose}
        isVisible
      />,
    );
    const filterButton = getByTestId('filterButton');
    const filtersChipsOptionButtonCategory2 = getByTestId(
      'filtersChipsOptionButtonCategory2',
    );
    fireEvent.press(filtersChipsOptionButtonCategory2);
    fireEvent.press(filterButton);

    expect(mockOnClose).toHaveBeenCalled();
    expect(mockSelectedOption.category).toBe('category 2');

    fireEvent.press(filtersChipsOptionButtonCategory2);
    fireEvent.press(filterButton);
    expect(mockSelectedOption.category).toBe(null);
  });

  it('should match snapshot', () => {
    expect(
      renderer
        .create(
          <ComponentWrapper
            yearOptions={mockYearOptions}
            categoryOptions={mockCategoryOptions}
            authorOptions={mockAuthorOptions}
            onClose={mockOnClose}
            isVisible
          />,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
