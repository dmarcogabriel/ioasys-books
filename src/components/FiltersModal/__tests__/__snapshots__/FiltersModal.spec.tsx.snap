// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FiltersModal should match snapshot 1`] = `
<View
  isVisible={true}
  onBackdropPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "padding": 16,
    }
  }
  testID="filtersModal"
>
  <RCTScrollView
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 4,
          "paddingBottom": 24,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": "100%",
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderColor": "#33333333",
              "borderRadius": 100,
              "borderStyle": "solid",
              "borderWidth": 1,
              "opacity": 1,
              "paddingBottom": 9,
              "paddingLeft": 9,
              "paddingRight": 9,
              "paddingTop": 9,
            }
          }
          testID="closeButton"
        >
          <Text>
            Icon
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 16,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 4,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
          testID="filtersChipsTitle"
        >
          Selecione a categoria
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 4,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
          testID="filtersChipsOptions"
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonCategory1"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextCategory1"
            />
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonCategory2"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextCategory2"
            />
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonCategory3"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextCategory3"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 16,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 4,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
          testID="filtersChipsTitle"
        >
          Selecione o ano
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 4,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
          testID="filtersChipsOptions"
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonYear1"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextYear1"
            >
              1992
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonYear2"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextYear2"
            >
              1993
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonYear3"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextYear3"
            >
              1994
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 16,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 12,
                "fontWeight": "bold",
                "marginBottom": 4,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
          testID="filtersChipsTitle"
        >
          Selecione o autor
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginBottom": 4,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 4,
              },
            ]
          }
          testID="filtersChipsOptions"
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonAuthor1"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextAuthor1"
            >
              Author 1
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonAuthor2"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextAuthor2"
            >
              Author 2
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#333333",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginBottom": 4,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 4,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filtersChipsOptionButtonAuthor3"
          >
            <Text
              isSelected={false}
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontSize": 12,
                  },
                ]
              }
              testID="filtersChipsOptionTextAuthor3"
            >
              Author 3
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#B22E6F",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 1,
              "justifyContent": "center",
              "marginBottom": 32,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 32,
              "opacity": 1,
              "paddingBottom": 8,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 8,
            }
          }
          testID="filterButton"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#B22E6F",
                  "fontSize": 16,
                },
              ]
            }
          >
            Filtrar
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
